src/include/migraphx/shape.hpp

```diff
@@ -131,6 +131,8 @@ struct shape
     shape with_lens(type_t t, const std::vector<std::size_t>& l) const;
     shape with_lens(const std::vector<std::size_t>& l) const;
 
+    shape with_type(type_t t) const;
```

Comment:
Can you rebase with develop and push changes again ? Paul's changes are showing up here. Otherwise, LGTM.